addons/mail/static/src/components/composer/composer.scss

```diff
@@ -25,6 +25,10 @@
     }
 }
 
+.o_ComposerView_button:hover {
+    background-color: var(--MessageActionView-background-color--hover, #{mix($border-color, $white)});
+}
+
```

Comment:
Just removing `bg-view` is good enough.
It's properly contrasty in dark theme. A bit soft in white theme, but actually it's best to keep slight hover style in the end.